### PR TITLE
allow port selection when you require self

### PR DESCRIPTION
**Pull request by anandkumarpatel**

### CODE REVIEW

**Comment by anandkumarpatel** [comment]
:eyes:

File: main.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+var publisher = require('./lib/publisher')();
+var noop = require('101/noop');
+
+var server;
+function start (port, cb) {
+  cb = cb || noop;
+  server = app.listen(port, function (err) {
+    if (err) { return cb(err); }
+    listener.start(publisher, cb);
+    debug('server listen on', process.env.PORT);
```
Comment (podviaznikov) [inline main.js]: use `port` param

Comment (anandkumarpatel) [inline main.js]: thanks!

File: main.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  cb = cb || noop;
+  server = app.listen(port, function (err) {
+    if (err) { return cb(err); }
+    debug('server listen on', port);
+    listener.start(publisher, cb);
+  });
+}
+function stop (cb) {
+  cb = cb || noop;
+  server.close(function (err) {
```
Comment (podviaznikov) [inline main.js:23]: `server` can be `null`. So you need to add check for that.
E.x. `stop` was called before `start` or `stop` was called twice.

Comment (anandkumarpatel) [inline main.js:23]: fixed

**Comment by podviaznikov** [comment]
nice! I like this refactoring.

**Pull request merged**